src/vs/vscode.proposed.d.ts

```diff
@@ -769,4 +769,23 @@ declare module 'vscode' {
 	}
 
 	//#endregion
+
+	//#region Terminal
+
+	export namespace window {
+		/**
+		 * The currently active terminals or an empty array.
+		 *
+		 * @readonly
+		 */
+		export let terminals: Terminal[];
+
+		/**
+		 * An [event](#Event) which fires when a terminal has been created, either through the
+		 * [createTerminal](#window.createTerminal) API or commands.
+		 */
+		export const onDidOpenTerminal: Event<Terminal>;
```

Comment:
Think about a `TerminalOpenEvent` so that you a chance of adding more stuff in the future..

Reply:
Then, should open close with close?

Reply:
`onDidCloseTerminal` already exists as an `Event<Terminal>`:

https://github.com/Microsoft/vscode/blob/da9720b39b1a912ce8009587731d824a890797ad/src/vs/vscode.d.ts#L4905-L4908

Reply:
missed that 🤦‍♂️

Reply:
Different file :wink: